### PR TITLE
 docs(config) update how to write the webpack configuration with typescript

### PR DESCRIPTION
**Pull request by youta1119**
issue #1735 
Update how to write the webpack configuration with typescript. Current configuration is wrong.
ts-node dosenot  support the module syntax of other than commonjs module. 
So, if module in compilerOptions in tsconfig.json is not commonjs, webpack getting error. like this.

```bash
(function (exports, require, module, __filename, __dirname) { import * as path from "path";
                                                              ^^^^^^

SyntaxError: Unexpected token import
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:588:28)
    at Module.m._compile (/Users/hogefuga/webpack-config-with-typescript/node_modules/ts-node/src/index.ts:392:23)

```
I have a two solution.
1. modify tsconfig.json
1. install tsconfig-paths

### First solution:

edit module in compilerOptions in tsconfig.json. please setting commonjs.

### Secoud solution:

first install tsconfig-paths

```bash
npm install --save-dev tsconfig-paths
```

**ts-node can resolve tsconfig.json using the environment variable provided by tsconfig-path.**

and setting environment variable `process.env.TS_NODE_PROJECT` provided by tsconfig-path like this.

```json
//tsconfig-for-webpack-config.json
{
  "compilerOptions": {
    "module": "commonjs"
  }
}

//package.json
{
  "scripts": {
      "build": "TS_NODE_PROJECT=\"tsconfig-for-webpack-config.json\" webpack"
  }
}
```
I make sample project. please see [this repository](https://github.com/youta1119/webpack-config-with-typescript)

### CODE REVIEW

File: src/content/configuration/configuration-languages.md [hunk 1]
```diff
@@ -39,6 +40,46 @@ const config: webpack.Configuration = {
 export default config;
 ```
 
+next check your tsconfig.json:
```
Comment (ChrisChinchilla) [inline src/content/configuration/configuration-languages.md]: @youta1119 Capital 'N' at the start of this line.

File: src/content/configuration/configuration-languages.md [hunk 2]
```diff
@@ -39,6 +40,46 @@ const config: webpack.Configuration = {
...
+
+If module in compilerOptions in tsconfig.json is commonjs, setting is complete, else webpack getting error.
```
Comment (ChrisChinchilla) [inline src/content/configuration/configuration-languages.md]: @youta1119 This is missing quite a few [articles](https://en.wikipedia.org/wiki/English_articles), for example…

> If **the** module in compilerOptions in tsconfig.json is commonjs, **the** setting is complete, else webpack will get an error.

Have a look through the rest of your PR and happy to help if you get stuck.

File: src/content/configuration/configuration-languages.md [hunk 3]
```diff
@@ -39,6 +40,46 @@ const config: webpack.Configuration = {
...
+This problem is occurring because ts-node does not support the module syntax of other than commonjs module.
+We have a two solutions.
+- modify tsconfig.json
+- install tsconfig-paths
+
+First solution:
+
+edit module in compilerOptions in tsconfig.json. please setting commonjs.
+
+Secoud solution:
```
Comment (ChrisChinchilla) [inline src/content/configuration/configuration-languages.md]: @youta1119 'Second'.

**Comment by youta1119** [comment]
@ChrisChinchilla Fix typo and few small language suggestions. Please review. 🙏

File: src/content/configuration/configuration-languages.md [hunk 4]
```diff
@@ -39,6 +40,47 @@ const config: webpack.Configuration = {
 export default config;
 ```
 
+Next check your tsconfig.json:
+
+If the module in compilerOptions in tsconfig.json is commonjs, the setting is complete, else webpack getting error.
```
Comment (dmitriid) [inline src/content/configuration/configuration-languages.md]:
```diff
- else webpack getting error.
+ else webpack will fail with an error
```

File: src/content/configuration/configuration-languages.md [hunk 5]
```diff
@@ -39,6 +40,47 @@ const config: webpack.Configuration = {
...
+This problem is occurring because ts-node does not support the module syntax of other than commonjs module.
+We have a two solutions.
+
+- modify tsconfig.json
+- install tsconfig-paths
+
+First solution:
+
+Edit module in compilerOptions in tsconfig.json. Please setting commonjs.
```
Comment (dmitriid) [inline src/content/configuration/configuration-languages.md]:
```diff
- Edit module in compilerOptions in tsconfig.json. Please setting commonjs.
+ In `tsconfig.json` look for `compilerOptions`. Set `target` to `"ES5"` and `module` to `"CommonJS"` (or completely remove the `module` option).
```

File: src/content/configuration/configuration-languages.md [hunk 6]
```diff
@@ -39,6 +40,47 @@ const config: webpack.Configuration = {
...
+
+First install tsconfig-paths
+
+```bash
+npm install --save-dev tsconfig-paths
+```
+
+T>ts-node can resolve tsconfig.json using the environment variable provided by tsconfig-path.
+
+and setting environment variable `process.env.TS_NODE_PROJECT` provided by tsconfig-path like this.
```
Comment (dmitriid) [inline src/content/configuration/configuration-languages.md]:
```diff
- and setting environment variable
+ Then set the environment variable
```

**Comment by youta1119** [comment]
@dmitriid 
Thank you for the review. I've fix it.

**Comment by youta1119** [comment]
@skipjack 
Thanks.  I think that good.
And I've fix markdown lint error. Please see it. 🙏

**Pull request merged**